### DISCUSSION TITLE
Add playback

### DISCUSSION THREAD

**Discussion by NJuste** [rerdavies/pipedal#73] [Ideas]
First, I want to thank you for this amazing work ! I'm using it on a rpi 4 with a hifi berry ADC DAC+ with my bass guitar and I just love it.

But one thing is missing me. I would like to add a music playback so I can practice at home. I have built a passiv stereo mixer so I can mix the output of pipedal with the standard output of the rpi on which I play the song I want. 

But I'm not happy with it because the sound is attenuated by the resistances. As I understand it, the sound card used by pipedal is dedicated to it. But is there a way to also play a song on that same audio card so the output plays both the sound of the instrument and the selected song ?

I hope I'm clear. I'm not a native English speaker.
Thanks !

**Comment by rerdavies** [comment]
You could try this: 

Build a patch with stereo inputs from hifi berry ADC. Connect the bass to one input channel, and a backing track audio device to the other (a phone would work well). In pipedal, add a split node after the input, and set the split type to LR (the left channel goes to the top rail; the right channel goes to the right rail). Add whatever bass effects you need to the appropriate rail, and then use the mixer controls on the LR split node to mix the output with the clean backing track audio on the other rail. The mixing controls are versatile enough to hard  pan the upper and lower rails to separate output channels, or mix them down to mono (center panned).

Not quite perfectly elegant, but it does what you want.

**Comment by NJuste** [comment]
Thank you for your answer. 
It's a good idea.  however, this will not allow music to be played on the rpi's sd card.  I may have a solution via the use of the alsa plugin "dmix" but I haven't had time to dig on this side yet.  I'll see.
